src/main/java/seedu/address/model/job/JobListManager.java

```diff
@@ -0,0 +1,113 @@
+package seedu.address.model.job;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.job.exceptions.DuplicateJobException;
+import seedu.address.model.job.exceptions.JobNotFoundException;
+
+
+public class JobListManager implements JobList {
+
+    private final ObservableList<Job> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Job> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    @Override
+    public boolean contains(Job toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::equals);
+    }
+
+    @Override
+    public void add(Job toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateJobException();
+        }
+        internalList.add(toAdd);
+    }
+
+    @Override
+    public void remove(Job toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new JobNotFoundException();
+        }
+    }
+
+    @Override
+    public void setJob(Job targetJob, Job editedJob) {
+        requireAllNonNull(targetJob, editedJob);
+
+        int index = internalList.indexOf(targetJob);
+        if (index == -1) {
+            throw new JobNotFoundException();
+        }
+
+        if (!targetJob.equals(editedJob) && contains(editedJob)) {
+            throw new DuplicateJobException();
+        }
+
+        internalList.set(index, editedJob);
+    }
+
+    @Override
+    public void setJobs(JobListManager replacement) {
+        requireNonNull(replacement);
+        internalList.setAll(replacement.internalList);
+    }
+
+    @Override
+    public void setJobs(List<Job> jobs) {
+        requireAllNonNull(jobs);
+        if (!jobsAreUnique(jobs)) {
+            throw new DuplicateJobException();
+        }
+        internalList.setAll(jobs);
+    }
+
+    /**
+     * Returns the backing list as an unmodifiable {@code ObservableList}.
+     */
+    public ObservableList<Job> asUnmodifiableObservableList() {
+        return internalUnmodifiableList;
+    }
+
+    /**
+     * Returns true if {@code jobs} contains only unique persons.
+     */
+    @Override
+    public boolean jobsAreUnique(List<Job> jobs) {
```

Comment:
this is O(n^2) :/

consider using something adding everything to a `HashSet` and either:
- comparing the size against the list size, or
- checking `Set.add` return value (this might be slightly more efficient if there are dupe jobs)

Reply:
Okay, I am unsure which comment I am exactly responding to here, apologies.

But with regards to the unique ID, I agree. Definitely, something we want to consider especially since Jobs may have identical fields. However, that requires editing the `Job` Class itself, which I think @Spyobird was working on, so I didn't want to modify it. Can add this up to the `Job` class. 

Wrt the Unique part, yeah. I pretty much refactored the parameters from the PersonsList, but a locally scoped HashSet does seem like a better idea, updating in next commit.